### PR TITLE
Fixes #3, allow latest and remove package_name

### LINKED ISSUE #3: Double whammy: allow 'latest' as a version, remove package_name
We should allow the user to specify `:latest` or `'latest'` as the version. We should also remove the `'package_name'` property from the `chef_ingredient` resource.

### PR DESCRIPTION
**Pull request by jtimberman**
Also cleans up rubocop and foodcritic warnings, because hey.

### CODE REVIEW

**Comment by jtimberman** [comment]
@chef-cookbooks/engineering-services Please review

File: libraries/helpers.rb [hunk 1]
```diff
@@ -19,12 +19,21 @@ module ChefIngredientCookbook
...
       if new_resource.respond_to?(:version)
-        product_lookup(product, new_resource.version)['ctl-command']
+        product_lookup(product, version_string(new_resource.version))['ctl-command']
       else
         product_lookup(product)['ctl-command']
       end
     end
 
+    def version_string(vers)
+      return '0.0.0' if vers == :latest || vers == 'latest'
```
Comment (schisamo) [inline libraries/helpers.rb]: Slightly cleaner version:

```ruby
return '0.0.0' if vers.to_sym == :latest
```

File: README.md [hunk 2]
```diff
@@ -40,7 +40,6 @@ A "chef ingredient" is the core package itself, or products or add-on components
 
 #### Properties
 - `product_name`: (name attribute) The product name. See the [PRODUCT_MATRIX.md](https://github.com/chef-cookbooks/chef-ingredient/blob/master/PRODUCT_MATRIX.md). For example, `chef-server`, `analytics`, `delivery`, `manage`, etc.
-- `package_name`: The name of the package in the repository. Should correspond to the published package names (`chef-server-core`, `opscode-manage`, etc).
 - `ctl_command`: The "ctl" command, e.g., `chef-server-ctl`. This should be automatically detected by the library helper method `chef_ctl_command`, but may need to be specified if something changes, like a new add-on is made available.
 - `options`: Options passed to the `package` resource used for installation.
 - `version`: Package version, e.g., `12.0.4`. Do not use if specifying `package_source`. Default `nil`.
```
Comment (schisamo) [inline README.md]: We need to document the usage of `:latest`.

File: libraries/helpers.rb (hunk 1 above)
Comment (jtimberman) [inline libraries/helpers.rb]: Duh :smile:

**Pull request merged**